blockchain/pool.go

```diff
@@ -292,7 +295,8 @@ func (pool *BlockPool) removePeer(peerID p2p.ID) {
 	for _, requester := range pool.requesters {
 		if requester.getPeerID() == peerID {
 			if requester.getBlock() != nil {
-				pool.numPending++
+				continue
```

Comment:
This is wrong?

Reply:
Is it ? We already have the block, so why should we increase numPending? I think this is why I was seeing ~8000 numPending.

AFAIK, numPending should track how many requests are open but still waiting for a block. Here we already have the block - so why would this be wrong?

Reply:
Because we're making the requester make the request again.
To be fair, this isn't the right place to do it, as there is a race condition.
I think maybe the requester should numPending... anyways, it should happen.

Reply:
why make the request again if we already got the block ? because presumably its a bad peer so we dont want to try its block ?

I think the reason numPending shoots then is because we dont clear the block here. so when we get the block again from a new peer and call `setBlock`, it returns false and we dont decrement.